src/Servant/Auth/Hmac/Crypto.hs

```diff
@@ -152,26 +168,30 @@ Authentication: HMAC <signature>
 It checks whether @<signature>@ is true request signature. Function returns 'Nothing'
 if it is true, and 'Just' error message otherwise.
 -}
-verifySignatureHmac
-    :: (SecretKey -> ByteString -> Signature)  -- ^ Signing function
-    -> SecretKey  -- ^ Secret key that was used for signing 'Request'
-    -> RequestPayload
-    -> Maybe LBS.ByteString
+verifySignatureHmac ::
+    -- | Signing function
+    (SecretKey -> ByteString -> Signature) ->
+    -- | Secret key that was used for signing 'Request'
+    SecretKey ->
+    RequestPayload ->
+    Maybe LBS.ByteString
 verifySignatureHmac signer sk signedPayload = case unsignedPayload of
-    Left err         -> Just err
-    Right (pay, sig) -> if sig == requestSignature signer sk pay
-        then Nothing
-        else Just "Signatures don't match"
+    Left err -> Just err
+    Right (pay, sig) ->
+        if sig == requestSignature signer sk pay
+            then Nothing
+            else Just "Signatures don't match"
   where
     -- Extracts HMAC signature from request and returns request with @authHeaderName@ header
     unsignedPayload :: Either LBS.ByteString (RequestPayload, Signature)
     unsignedPayload = case extractOn isAuthHeader $ rpHeaders signedPayload of
         (Nothing, _) -> Left "No 'Authentication' header"
         (Just (_, val), headers) -> case BS.stripPrefix "HMAC " val of
-            Just sig -> Right
-                ( signedPayload { rpHeaders = headers }
-                , Signature sig
-                )
+            Just sig ->
+                Right
+                    ( signedPayload{rpHeaders = headers}
+                    , Signature sig
+                    )
             Nothing -> Left "Can not strip 'HMAC' prefix in header"
 
```

Comment:
I don't know much about the HMAC authentication standard, but I think this code makes the library can not be reused in many cases :
- The signature is specified in a different header name (such as [x-sendbird-signature)](https://sendbird.com/docs/chat/v3/platform-api/guides/webhooks#2-headers-3-x-sendbird-signature) 
- The payload is request body or combination of request body with some header content (such as [Stripe payload](https://stripe.com/docs/webhooks/signatures#verify-manually))

So I think it could be better if we can specify the header name of the signature and the function to map from request to singed payload

Reply:
Like you said, our current implementation of the authentication differs from Sendbird's which also differs from Stripe's. While it would be interesting to be customizable to handle as much use case as possible, I fear that there's too many different use cases out there and we would lose the "simple" aspect of the library.

For example, we could expose the `requestSignature` function and allow the client to change the actual implementation, but the library may lose most of its usefulness as a consequence.

What's everyone else's opinion on that?

Reply:
Sorry, missed this earlier.  
@thongpv87 Just so that I understand your point better : 
- Does it mean that with this change merged into `master`, that you would have to re-implement or change some existing code in order to be compatible with a future version of `servant-hmac-auth`?

Reply:
This PR is essentially formatting and testing so there should be no impact on existing projects.
#57 may be more impactful, but the previous API should be untouched (the tests from this PR are actually testing the previous API and there weren't broken in the #57).